src/errors/EmptyStringError.ts

```diff
@@ -0,0 +1,12 @@
+import { BaseIsomerError } from "./BaseError"
+
+export default class EmptyStringError extends BaseIsomerError {
+  constructor(
+    message = "An empty string was provided for a method that requires a non-empty string"
+  ) {
+    super()
+    Error.captureStackTrace(this, this.constructor)
+    this.name = this.constructor.name
+    this.message = message
```

Comment:
i think we'd still need to preserve `this.name = this.constructor.name` because if it's at `super`, the constructor will be `BaseIsomerError`. not entirely sure on this, wdyt?

Reply:
woah my hunch was that you were right, but turns out no sia 
The way I validated this was i commented out 
`// this.name = this.constructor.name` and placed a `console.log(this.name)` right below, and it magically printed out `EmptyStringError` rather than `BaseIsomerError` 

At the same time, `this.message = message` in line 10 should still exist!

Reply:
huh cool! so to double check - we don't need the `this.name = this.constructor.name`?

Reply:
yes sir

Reply:
done!